packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.test.js

```diff
@@ -0,0 +1,191 @@
+    references: [
+      {
+        refType: 'CHILD_OF',
+        span: {
+          spanID: parentSpanID,
+          references: [
+            {
+              refType: 'CHILD_OF',
+              span: {
+                spanID: rootSpanID,
+              },
+            },
+          ],
+        },
+      },
+    ],
+  };
+  const specialRootID = 'root';
+  let props;
+  let wrapper;
+
+  beforeEach(() => {
+    props = {
+      addSpanId: jest.fn(),
+      hoverSpanIds: new Set(),
+      removeSpanId: jest.fn(),
+      span: spanWithTwoAncestors,
+    };
+    wrapper = shallow(<UnconnectedSpanTreeOffset {...props} />);
+  });
+
+  describe('.SpanTreeOffset--indentGuide', () => {
+    it('should only render one .SpanTreeOffset--indentGuide for entire trace if span has no ancestors', () => {
+      props.span = {
+        spanID: 'parentlessSpanID',
+        references: [
+          {
+            refType: 'NOT_CHILD_OF',
+            span: {
+              spanID: 'notAParentSpanID',
+              references: [],
+            },
+          },
+        ],
+      };
+      wrapper = shallow(<UnconnectedSpanTreeOffset {...props} />);
+      const indentGuides = wrapper.find('.SpanTreeOffset--indentGuide');
+      expect(indentGuides.length).toBe(1);
+      expect(indentGuides.prop('data--ancestor-id')).toBe(specialRootID);
+    });
+
+    it('should render one .SpanTreeOffset--indentGuide per ancestor span, plus one for entire trace', () => {
+      const indentGuides = wrapper.find('.SpanTreeOffset--indentGuide');
+      expect(indentGuides.length).toBe(3);
+      expect(indentGuides.at(0).prop('data--ancestor-id')).toBe(specialRootID);
+      expect(indentGuides.at(1).prop('data--ancestor-id')).toBe(rootSpanID);
+      expect(indentGuides.at(2).prop('data--ancestor-id')).toBe(parentSpanID);
+    });
+
+    it('should add .activeMouseover className to correct indentGuide', () => {
+      props.hoverSpanIds = new Set([parentSpanID]);
+      wrapper = shallow(<UnconnectedSpanTreeOffset {...props} />);
+      const activeMouseoverIndentGuide = wrapper.find('.activeMouseover');
+      expect(activeMouseoverIndentGuide.length).toBe(1);
+      expect(activeMouseoverIndentGuide.prop('data--ancestor-id')).toBe(parentSpanID);
+    });
+
+    it('should call props.addSpanId on mouse enter', () => {
+      wrapper.find({ 'data--ancestor-id': parentSpanID }).simulate('mouseenter', {});
+      expect(props.addSpanId).toHaveBeenCalledTimes(1);
+      expect(props.addSpanId).toHaveBeenCalledWith(parentSpanID);
+    });
+
+    it('should not call props.addSpanId on mouse enter if mouse came from a indentGuide with the same ancestorId', () => {
+      wrapper.find({ 'data--ancestor-id': parentSpanID }).simulate('mouseenter', {
+        relatedTarget: {
+          getAttribute: () => parentSpanID,
+        },
+      });
+      expect(props.addSpanId).not.toHaveBeenCalled();
+    });
+
+    it('should call props.removeSpanId on mouse leave', () => {
```

Comment:
Btw, I adopted describing tests cases with descriptions like

```
it('calls props.removeSpanId on mouse leave', () => {});
// instead of the longer form
it('should call ...')
```
It's a bit shorter :)